Fix roster submission

katsukixyz opened the pull request:
## Roster submission persists to database

- Refactored `route_selection` page to fetch from routes instead of schools, navigating to `bus_checkin_roster/route` only displays students that are assigned to that bus route to reflect this change
- Student checkIns are saved to the database and are not cleared after refresh (specific to the day)
- Bus routes are marked as complete/incomplete based on the most recent roster submission and whether that corresponds to the client's current date
- Upon completing a route, relevant tables and buttons are grayed out to prevent further editing
- Submission of roster results in success/failure popup (required refactoring `modal.jsx` to allow modal to close upon form submission)

### Related Github Issues

- #44

kphan32 reviewed (changes requested):
Testing locally, student notes and roster submissions do not appear to persist.

katsukixyz commented:
Refactored attendance API to use ObjectId from MongoDB instead of relying on the `studentID` field in the `students` collection. `checkInTimes` and `notes` fields were also removed and replaced with a `checkIns` array that stores student notes by date. 

![image](https://user-images.githubusercontent.com/31021209/138372265-df4b664c-7505-4179-bce2-9e3e8a039f9b.png)

kphan32 reviewed (approved):
lgtm :+1:
